### PR TITLE
Boolean Support

### PR DESCRIPTION
**Pull request by apavlo**
This PR adds support for booleans. I have not tested this through the SQL interface.

It also adds some additional EE tests as an attempt to increase our coverage. There is a ASAN problem with `ErrorReporterTest` (see #610) so that test case is disabled for now.

Shout out to @thepinetree and @pmenon for helping me with this.

https://youtu.be/MxCDcJlXOIU

### CODE REVIEW

**Comment by mbutrovich** [comment]
Added myself as a reviewer so I could push an update fixing the merge conflict.

File: src/include/network/postgres/postgres_packet_writer.h [hunk 1]
```diff
@@ -9,6 +9,17 @@
 #include "planner/plannodes/output_schema.h"
 
 namespace terrier::network {
+
+/**
+ * The string value to use for 'true' boolean values
+ */
+#define POSTGRES_BOOLEAN_STR_TRUE "t"
```
Comment (mbutrovich) [inline src/include/network/postgres/postgres_packet_writer.h]: prefer `constexpr char blah[] = "blah"` over global `#define`.

Comment (apavlo) [reply src/include/network/postgres/postgres_packet_writer.h]: Fixed.

**Comment by mbutrovich** [comment]
Benchmark regression is unrelated.

**Pull request merged**